src/main/java/appeng/me/cache/NetworkMonitor.java

```diff
@@ -149,9 +173,186 @@ protected void postChange( final boolean add, final Iterable<T> changes, final B
 		}
 
 		final NetworkMonitor<?> last = DEPTH.pop();
+		semaphore--;
 		if( last != this )
 		{
 			throw new IllegalStateException( "Invalid Access to Networked Storage API detected." );
 		}
 	}
+
+	private void updateCache( Iterable<T> changes, boolean add )
+	{
+		for( final T changedItem : changes )
+		{
+			T difference = changedItem;
+
+			if( !add && changedItem != null )
+			{
+				( difference = changedItem.copy() ).setStackSize( -changedItem.getStackSize() );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract the assignment out of this expression. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)